.github/workflows/dev-setup.yml

```diff
@@ -41,6 +41,12 @@ jobs:
     timeout-minutes: 30
     if: github.event_name != 'schedule' || github.repository == 'mlflow-automation/mlflow'
     steps:
+      - name: Free Disk Space (Ubuntu)
```

Comment:
is this change related to the fix? if not, would suggest to separate this to a different PR so that we can keep the blame clean. i.e. later if we debug and try to find when and why certain change was introduced, we could trace back to the PR with right context. cuz this PR's description didn't explain why we made this free disk space change.

Reply:
without this, the CI task failed with "No space left" : https://github.com/WeichenXu123/mlflow/actions/runs/9677893419/job/26700509236#step:4:2084